File: 프로그래머스/0/120902. 문자열 계산하기/문자열 계산하기.py
def solution(my_string):
    answer = int(my_string.split(" ")[0])
    box = ["0"]
    for i in my_string.split(" ")[1:]:
        box.append(i)
        if i.isdigit() and box[-2] == "-":
            answer -= int(i)
            box = ["0"]
        elif i.isdigit() and box[-2] == "+":
            answer += int(i)
            box = ["0"]
    return answer

File: 프로그래머스/0/120902. 문자열 계산하기/문자열 계산하기.java
class Solution {
    public int solution(String my_string) {
    int answer = 0;
    String [] box = new String[]{"+"};

    for (String i : my_string.split(" ")){

        if (!i.matches("[0-9]+")){
            box[0] = i;
        }

        else {
            if (box[0].equals("+")){
                answer += Integer.parseInt(i);
            }

            else {
                answer -= Integer.parseInt(i);
            }
        }
    }
    return answer;
}
}